### PR TITLE
PRO-2234 Bypassing lockout timeout

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-2234/pillar-wallet-bypassing-lockout-timeout

## Description

- Fixed bypassing lockout timeout
- Used WorldTime API

## Motivation and Context

- https://hackerone.com/reports/2354251

## How Has This Been Tested?

- Tested on Simulator

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/actions/appActions.js [hunk 1]
```diff
@@ -187,11 +189,21 @@ export const initAppAndRedirectAction = () => {
...
+          return { data: null };
+        });
+
       const { failedAttempts = {}, pinAttempt = {} } = get(storageData, 'pinAttempt', {});
-      const { numberOfFailedAttempts = 0, date = new Date() } = failedAttempts;
+      const { numberOfFailedAttempts = 0, date = data?.datetime || new Date() } = failedAttempts;
       const { pinAttemptsCount = 0 } = pinAttempt;
 
-      if (new Date(date)?.toDateString() === new Date()?.toDateString()) {
+      const today = data?.datetime ? new Date(data.datetime)?.toDateString() : new Date()?.toDateString();
```
Comment (IAmKio) [inline src/actions/appActions.js:205]: did we need the ? on the `new Date()?.toDateString()`?

**Review by IAmKio — COMMENTED** [review]
LGTM but what does the UI look like when no internet is available? I saw you added the flag to `isLoading` flag, does it just display a spinner?

File: src/actions/appActions.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/actions/appActions.js:205]: did we need the ? on the new Date()?.toDateString()?
Not mandatory, I'll remove it. 

I saw you added the flag to isLoading flag, does it just display a spinner?

https://github.com/pillarwallet/pillarwallet/assets/82652040/584cb0a9-7178-4c7d-b5b0-bd68532e7811

Comment (IAmKio) [reply src/actions/appActions.js:205]: @lalitkumarrock thanks. Does pressing any digits of the keypad have any effect here? Just trying to close all possible attack vectors

Comment (lalitkumarrock) [reply src/actions/appActions.js:205]: At that time not working any keypad.

https://github.com/pillarwallet/pillarwallet/assets/82652040/d8cb08b0-9a55-4998-a36f-a65f07c66d0a

**Review by IAmKio — APPROVED** [review]

**Pull request merged**